src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -88,7 +108,18 @@ public abstract class StringConfigurationSource : IConfigurationSource
         /// <returns><see cref="ConcurrentDictionary{TKey, TValue}"/> containing all of the key-value pairs.</returns>
         public IDictionary<string, string> GetDictionary(string key)
         {
-            return ParseCustomKeyValues(GetString(key));
+            return ParseCustomKeyValues(GetString(key), false);
```

Comment:
```suggestion
            return ParseCustomKeyValues(GetString(key), allowOptionalMappings: false);
```

Reply:
Done in c1ac5ce1868b7c359e741bd04d05e742647a9a33